src/main/java/me/pheric/pcore/database/Database.java

```diff
@@ -0,0 +1,171 @@
+package me.pheric.pcore.database;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Manages and connects to a MySQL database
+ *
+ * @author StealthMajr (stealthmajr@gmail.com)
+ * @see Connection
+ * @see HikariDataSource
+ * @see Result
+ * @see DB
+ * @since 0.1.0
+ */
+public class Database {
+
+    private HikariDataSource dataSource;
+
+    /**
+     * Initializes the class
+     *
+     * @param jdbcUrl  The JDBC URL for connecting, usually in the following format:
+     *                 jdbc:mysql://hostname:port/database
+     * @param username The username of this MySQL connection
+     * @param password The password of this MySQL connection
+     */
+    public Database(String jdbcUrl, String username, String password) {
```

Comment:
I'd like an overload in case other options are necessary, such as specifying the port.

Reply:
Definitely! Can do. I'll most likely be allowing multiple database types, so you'll be able to use PostgreSQL, and others can use MySQL if needed.

Reply:
Port is already being specified by the jdbcUrl. No need to add an overload for that.

Reply:
Whoops, good point. It also looks like I can add my own database with `setGlobal()`